### PR TITLE
[Feature][File connector] Support datahub sink 

### PR DESCRIPTION
**Pull request by chessplay**
** Support datahub sink #1946 

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-connectors-v2/connector-datahub/pom.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      <groupId>org.apache.seatunnel</groupId>
+      <artifactId>connector-common</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+
+    <dependency>
+      <groupId>com.aliyun.datahub</groupId>
+      <artifactId>aliyun-sdk-datahub</artifactId>
+      <version>2.19.0-public</version>
+    </dependency>
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-datahub/pom.xml]: Hi, Version should be managed uniformly in the root pom

File: seatunnel-connectors-v2/connector-datahub/src/main/java/org/apache/seatunnel/connectors/seatunnel/sink/DataHubWriter.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+ * Datahub write class
+ */
+public class DataHubWriter extends AbstractSinkWriter<SeaTunnelRow, Void> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DataHubWriter.class);
+    private DatahubClient dataHubClient;
+    private String project;
+    private String topic;
+    private SeaTunnelRowType seaTunnelRowType;
+    private static final int DATAHUB_CONNECT_MAX_TIMEOUT = 10000;
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-datahub/src/main/java/org/apache/seatunnel/connectors/seatunnel/sink/DataHubWriter.java]: Is it possible to give this parameter to the user as an optional configuration?

File: seatunnel-connectors-v2/connector-datahub/src/main/java/org/apache/seatunnel/connectors/seatunnel/sink/DataHubWriter.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            PutRecordsResult result = dataHubClient.putRecords(project, topic, recordEntries);
+            LOG.info("putRecordResult:" + result.toString());
+        }  catch (DatahubClientException e) {
+            LOG.error("requestId:" + e.getRequestId() + "\tmessage:" + e.getErrorMessage());
+        }
+    }
+
+    @Override
+    public void close() throws IOException {
+
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-datahub/src/main/java/org/apache/seatunnel/connectors/seatunnel/sink/DataHubWriter.java]: Does the Client need to be closed?

File: seatunnel-connectors-v2/connector-datahub/src/main/java/org/apache/seatunnel/connectors/seatunnel/sink/DataHubWriter.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        RecordSchema recordSchema = dataHubClient.getTopic(project, topic).getRecordSchema();
+        for (int i = 0; i < fieldNames.length; i++) {
+            TupleRecordData data = new TupleRecordData(recordSchema);
+            data.setField(fieldNames[i], fields[i]);
+            RecordEntry recordEntry = new RecordEntry();
+            recordEntry.setRecordData(data);
+            recordEntries.add(recordEntry);
+        }
+        try {
+            PutRecordsResult result = dataHubClient.putRecords(project, topic, recordEntries);
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-datahub/src/main/java/org/apache/seatunnel/connectors/seatunnel/sink/DataHubWriter.java]: I'm not sure if I need to judge the corresponding result code.

File: plugin-mapping.properties [hunk 5]
```diff
@@ -111,7 +111,6 @@ seatunnel.source.HdfsFile = connector-file-hadoop
 seatunnel.sink.HdfsFile = connector-file-hadoop
 seatunnel.source.LocalFile = connector-file-local
 seatunnel.sink.LocalFile = connector-file-local
-seatunnel.source.OssFile = connector-file-oss
```
Comment (hailin0) [inline plugin-mapping.properties]: Why remove?

File: seatunnel-examples/seatunnel-flink-connector-v2-example/pom.xml [hunk 6]
```diff
@@ -76,6 +76,11 @@
             <artifactId>connector-dingtalk</artifactId>
             <version>${project.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>connector-datahub</artifactId>
+            <version>${project.version}</version>
+        </dependency>
```
Comment (hailin0) [inline seatunnel-examples/seatunnel-flink-connector-v2-example/pom.xml]: Would you like to add an `DataHub` example?

reference
https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-examples/seatunnel-flink-connector-v2-example/src/main/java/org/apache/seatunnel/example/flink/v2
https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-examples/seatunnel-spark-connector-v2-example

**Review by hailin0 — COMMENTED** [review]
Can you add connector-v2 e2e-testcase?

reference：
https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-flink-connector-v2-e2e
https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-spark-connector-v2-e2e

**Pull request closed without merging**